python/ql/test/library-tests/PointsTo/new/code/a_simple.py

```diff
@@ -50,3 +50,6 @@ def multi_assign_and_packing(a, b="b", c="c"):
     g
     h
     i
+    l, m = (1,) + (2,)
+    l
+    m
```

Comment:
I'm assuming that this is the test for the points-to of the RHS being `UnknownInstanceInternal`? If so, I would like an example that would still hold once we teach our analysis that the result of `(1,) + (2,)` is `(1,2)` :smile: 

I was going to suggest just using `l, m = a`, but I'm not sure if that would work? :confused:

Reply:
I don't mean to burst your bubble, but I think it's unlikely that we'll teach it to add tuples. If we did, we would run into nontermination on things like 
```python
x = (1,2)
while True:
    x = x + x
```
We _could_ do a special case when the two parts are literal tuples, but this will have to depend on whether there's any benefit in doing so.

You raise a valid point about `l, m = a`. I think this one won't work, since `a` (should) be `UnknownInternal`. I'm thinking instead we should be using the `subscriptUnknown` method, since the multiassignment _could_ be seen as a shorthand for multiple assignments using subscripting. Then `l, m = a` should work again.

More generally, I'm less and less convinced that it makes sense to exclude the `Unknown` objects from being `Value`s. I'll make an internal issue to continue the discussion on this.